external/anomaly/ote_anomalib/callbacks/progress.py

```diff
@@ -112,6 +112,15 @@ def _get_progress(self, stage: str = "train") -> float:
             raise ValueError(f"Unknown stage {stage}. Available: train, predict and test")
 
         return self._progress
+    
+    def on_validation_epoch_end(self, trainer, pl_module):
+        super().on_validation_epoch_end(trainer, pl_module)
+        score = None
+        metric = getattr(self.update_progress_callback, 'metric', None)
+        if metric in trainer.logged_metrics:
+            score = float(trainer.logged_metrics[metric])
+        progress = int(self._get_progress('predict'))
+        self.update_progress_callback(progress=progress, score=score)
```

Comment:
Not sure why `update_progress_callback` is called from here. We only update progress for train, test or predict. I feel this will have unintended consequences such as progress jumping between x%-100%. Especially when  Has this been tested in the UI? We removed val step just to avoid this issue. https://github.com/openvinotoolkit/training_extensions/commit/95126f6fd6731f89921731ba901d1e150c72aca0
In case this works in the UI then I am fine with it.

Reply:
Yes I understood your comment. We checked again and we found that if we call "update_progress_callback" in the function "on_train_epoch_end", the score comes after validation step end.
So how about replacing "on_validation_epoch_end" to "on_train_epoch_end" and pass "train" as argument of "_get_progress"?
We think it will work well in both hpopt and UI.

Reply:
@SangdaeNam, thank you for the PR. 

I have a question: can you tell me why you want to add hpo update to the `ProgressCallback`? Unless I miss something here, I feel like HPO and ProgressBar serve for different purposes. 

Would it be an idea to create a separate callback for HPO itself? If we have a HPO callback, we could add it to the list of callbacks in the training [here](https://github.com/openvinotoolkit/training_extensions/blob/develop/external/anomaly/ote_anomalib/train_task.py#L53)

And our callbacks, would become
```
callbacks = [
    ProgressCallback(parameters=train_parameters), 
    MinMaxNormalizationCallback(), 
    HPOCallback(),
]
```

Am I missing something? Can you please clarify?

Reply:
Hi Samet, thank you for the comment!
HPO actually utilizes ote_sdk as much as possible for using common interface for all tasks.
HPOCallback is inherited from UpdateProgressCallback class as one of this direction.
For this reason, HPO uses ProgressCallback by being passed as attribute of "train_parameters" argument for reporting.
This usage is also commented in UpdateProgressCallback class.
If you have any concerns to use ProgressCallback, please let me know about that.
We'll try our best to resolve that concerns.

Reply:
We pushed commit as we said above.
Could you comment for this modification, @samet-akcay @ashwinvaidya17?

Reply:
Can you point me to the code for HPOCallback? I still feel that HPOCallback should be a separate thing. Now, if we use HPOCallback on metrics from train_epoch_end, we will not be using val set for hyperparameter optimization. And I feel that we might overfit the params on the train set. If we decouple HPOCallback and Progress callback then we can safely add val_epoch_end to the model without messing with the progress on the UI.